cc_library/src/scripts/controlboard.py

```diff
@@ -12,53 +13,55 @@
 
 
 class ControlBoard(Device):
-    GPIO.setmode(GPIO.BCM)
-    """
-    Define pin numbers to which units are connected on Pi.
-    """
-    redSwitch = 27
-    orangeSwitch = 22
-    greenSwitch = 18
-    mainSwitch = 23
-    switches = [redSwitch, orangeSwitch, greenSwitch, mainSwitch]
-
-    redLight1 = 9
-    redLight2 = 15
-    redLight3 = 17
-    greenLight1 = 10
-    greenLight2 = 14
-    greenLight3 = 4
-
-    redLEDs = [redLight1, redLight2, redLight3]
-    greenLEDs = [greenLight1, greenLight2, greenLight3]
-
-    a_pin0 = 24
-    a_pin1 = 25
-    a_pin2 = 8
-    b_pin0 = 7
-    b_pin1 = 1
-    b_pin2 = 12
-
-    GPIO.setup(redSwitch, GPIO.IN, pull_up_down=GPIO.PUD_UP)
-    GPIO.setup(orangeSwitch, GPIO.IN, pull_up_down=GPIO.PUD_UP)
-    GPIO.setup(greenSwitch, GPIO.IN, pull_up_down=GPIO.PUD_UP)
-    GPIO.setup(mainSwitch, GPIO.IN, pull_up_down=GPIO.PUD_UP)
-
-    GPIO.setup(redLight1, GPIO.OUT)
-    GPIO.setup(redLight2, GPIO.OUT)
-    GPIO.setup(redLight3, GPIO.OUT)
-    GPIO.setup(greenLight1, GPIO.OUT)
-    GPIO.setup(greenLight2, GPIO.OUT)
-    GPIO.setup(greenLight3, GPIO.OUT)
-
-    adc = Adafruit_ADS1x15.ADS1115()
-
-    GPIO.setup(a_pin0, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
-    GPIO.setup(a_pin1, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
-    GPIO.setup(a_pin2, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
-    GPIO.setup(b_pin0, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
-    GPIO.setup(b_pin1, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
-    GPIO.setup(b_pin2, GPIO.IN, pull_up_down=GPIO.PUD_DOWN)
+
+    def __init__(self):
+        Device.__init__(self)
+        self.adc = ADC()
+        self.GPIO.setmode(GPIO.BCM)
```

Comment:
I'm going to remove self in self.GPIO.setmode(GPIO.BCM) because it is not suppose to be there

Reply:
good idea